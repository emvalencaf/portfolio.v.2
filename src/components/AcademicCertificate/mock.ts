import { AcademicCertificateProps } from ".";

export default {
	courseTitle: "Curso de React.Js e Next.Js (Intermediário e Avançado)",
	institution: "Udemy, Instrutor: Luiz Otávio Miranda",
	courseWork: "93 horas",
	startIn: 2932983,
	endIn: 3932983,
	urlDownload: "/assets/files/UC-44225967-b7ec-4add-8983-426364ad7bb6.pdf"
} as AcademicCertificateProps;