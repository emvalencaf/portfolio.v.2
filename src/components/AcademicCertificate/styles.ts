import styled, { css, DefaultTheme } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';

type WrapperAcademicCertificateProps = {
	theme: DefaultTheme;
	showDetails: boolean;
}

const changeCursor = () => css`
	cursor: pointer;
`;

export const Wrapper = styled.div<WrapperAcademicCertificateProps>`
	${({ theme, showDetails }) => css`
	padding: 2rem;
	${!showDetails && changeCursor()}

	& ${HeadingContainer} {
		margin: 0px;
	}
	`}
`;

export const HeaderAcademicCertificate = styled.div`
	${({ theme }) => css`
		display: flex;
		align-items: center;
		justify-content: center;
		margin: ${theme.spacings.small} auto;
	`}
`;

export const BodyAcademicCertificate = styled.div`
	${({ theme }) => css`
		display: flex;
		flex-direction: column;
		align-items: center;
		justify-content: center;
		gap: ${theme.spacings.small};
		margin: ${theme.spacings.small} auto;
	`}
`;