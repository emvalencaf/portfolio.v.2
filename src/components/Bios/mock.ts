import { BiosProps } from ".";

export default {
	bios: `
<p>
	Lorem ipsum dolor sit amet consectetur adipisicing elit. Inventore rem minus dolore ducimus quos, sed dolores odit maiores quam tempora atque deserunt voluptas perferendis, aspernatur, sequi dignissimos reiciendis. In, esse!
	Lorem ipsum, dolor sit amet consectetur adipisicing elit. Non a deleniti quia. Magnam aspernatur cumque, nemo vero libero excepturi consequuntur corporis ullam nobis molestiae? Atque nam praesentium ullam quaerat quisquam?
</p>
<p>
	Lorem ipsum dolor sit amet consectetur adipisicing elit. Optio cumque recusandae libero sequi, tempora iste, illo, possimus unde cum numquam quae corrupti minima tenetur mollitia earum repellendus animi molestiae voluptates.
	Lorem ipsum dolor sit, amet consectetur adipisicing elit. Obcaecati dolores fugiat ipsam ut odit sed aliquid ducimus molestiae vero ipsa nam veniam dolor libero labore qui pariatur, alias assumenda itaque?
	Lorem ipsum dolor sit amet, consectetur adipisicing elit. Illo cum temporibus nesciunt sequi odit ipsam harum magnam dolore a molestiae. Totam debitis, rerum aliquid sit non ratione. Et, omnis iusto!
</p>
<p>
	Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nam dicta aut cum sequi voluptatibus. Vel pariatur veritatis quidem. Culpa vero deleniti minima earum quod, odit deserunt fugit reprehenderit architecto in!
</p>
	`,
	profilePhoto: {
		altText: "profile picture",
		srcImg: "assets/images/potrait-photo.jpg",
	},
} as BiosProps;
