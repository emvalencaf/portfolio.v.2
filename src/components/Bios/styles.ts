import styled, { css } from "styled-components";
export const Wrapper = styled.div`
	${({ theme }) => css`
		@media ${theme.media.lteMedium} {
			flex-direction: column;
		}

		display: flex;
		flex-direction: row;
		flex-wrap: wrap;
		justify-content: space-between;
		align-items: center;
		gap: ${theme.spacings.xhero};
		text-align: justify;
		& p {
			text-align: justify;
			padding: ${theme.spacings.small};
			max-width: 550px;
		}
	`}
`;

export const ContentContainer = styled.div`
	${() => css`
		max-height: 350px;
		overflow-y: auto;
	`}
`;
