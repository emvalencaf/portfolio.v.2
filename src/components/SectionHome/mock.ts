// type
import { SectionHomeProps } from ".";

export default {
	id: "#",
	ownerName: "Tício de Souza e Silva",
	ocupation: "Desenvolvedor WEB Júnior",
	mainStack: [
		"Javascript",
		"NodeJS",
		"React",
	],
} as SectionHomeProps;
