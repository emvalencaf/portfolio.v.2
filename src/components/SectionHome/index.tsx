// components
import Heading from '../Heading';
import Section from '../Section';

// styles
import * as Styled from './styles';

// types
export type SectionHomeProps = {
	id: string;
	ownerName?: string;
	ocupation?: string;
	mainStack?: string[];
};

const SectionHome = ({ id, ownerName, ocupation, mainStack = [] }: SectionHomeProps) => {
	return (
		<Styled.Wrapper id={id}>
			<Section>
				<p>
					Olá, seja bem-vindo ao meu portfólio.
				</p>
				{ownerName && (
					<Heading as="h1" size="big">
						{ownerName}.
					</Heading>
				)}
				{ocupation && (
					<Heading as="h2" size="big">
						{ocupation}.
					</Heading>
				)}
				{mainStack.length > 0 && (
					<Heading as="h2" size="medium">
						{mainStack.join(" | ")}
					</Heading>
				)}
			</Section>
		</Styled.Wrapper>
	);
};

export default SectionHome;
