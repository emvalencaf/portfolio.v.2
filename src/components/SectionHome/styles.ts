import styled, { css } from "styled-components";
import { Title as HeadingContainer } from "../Heading/styles";

export const Wrapper = styled.div`
	${({ theme }) => css`
		display: flex;
		flex-direction: row;
		align-items: center;
		justify-content: space-between;
		height: 600px;
		width: 100%;
		margin: 0px auto;

		background-position: center;
		background-repeat: no-repeat;
		background-size: cover;

		& p {
			margin: ${theme.spacings.small};
			padding: 0;
			width: 100%;
			font-size: ${theme.fonts.sizes.medium};
		}

		& ${HeadingContainer}:first-child {
			text-shadow: 0px 0px 20px ${theme.colors.quaternary};
		}

		& ${HeadingContainer} {
			margin: ${theme.spacings.small};
			padding: 0;
			width: 100%;
			text-shadow: 0px 0px 20px ${theme.colors.senary};
		}
		& ${HeadingContainer}::after {
			content: none;
		}
	`}
`;

export const IntroContainer = styled.div`
	${() => css`
		display: flex;
		flex-direction: column;
		align-items: center;
		justify-content: center;
	`}
`;

export const GithubContainer = styled.div`
	${({ theme }) => css`
		@media ${theme.media.gteOrEqLarge} {
			display: flex;
			flex-direction: column;
			align-items: center;
			justify-content: center;
			max-width: 300px;
			& ${HeadingContainer} {
				margin: 0px;
			}
		}
		display: none;
	`}
`;
