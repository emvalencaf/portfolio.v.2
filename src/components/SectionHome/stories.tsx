import { Meta, Story } from "@storybook/react/types-6-0";
import SectionHome, { SectionHomeProps } from ".";
import mock from "./mock";

export default {
	title: "SectionHome",
	component: SectionHome,
	args: mock,
	parameters: {
		layout: "fullscreen",
	},
} as Meta<SectionHomeProps>;

export const Template: Story<SectionHomeProps> = (args) => {
	return (
		<div>
			<SectionHome {...args} />
		</div>
	);
};
