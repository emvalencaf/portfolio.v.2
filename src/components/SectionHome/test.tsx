import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';

// react component to be test
import SectionHome from '.';

// mock
import mock from './mock';

describe('<SectionHome />', () => {
	it('should render component it props default values', () => {
		renderTheme(<SectionHome {...mock} />);

		expect(screen.getByRole('heading', { name: 'Oi' })).toBeInTheDocument();
	});
});
