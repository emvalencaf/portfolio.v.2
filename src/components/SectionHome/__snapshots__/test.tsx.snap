// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionHome /> should render match a snapshot 1`] = `
.c3 {
  color: #0A1128;
  line-height: 1.2;
  font-size: 4.0rem;
  text-transform: none;
}

.c4 {
  color: #0A1128;
  line-height: 1.2;
  font-size: 3.2rem;
  text-transform: none;
}

.c1 {
  height: 500px;
  max-height: 500px;
  width: 100%;
  padding: 0px 6.4rem;
  margin: calc(6.4rem + 5.6rem) auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  margin: 0 auto;
}

.c0 p {
  margin: 1.6rem;
  padding: 0;
  width: 100%;
  font-size: 2.4rem;
}

.c0 .c2 {
  margin: 1.6rem;
  padding: 0;
  width: 100%;
}

@media (max-width:360px) {
  .c3 {
    font-size: 2.4rem;
  }
}

@media (max-width:768px) {
  .c3 {
    font-size: 3.2rem;
  }
}

@media (max-width:360px) {
  .c4 {
    font-size: 1.6rem;
  }
}

@media (max-width:768px) {
  .c4 {
    font-size: 2.4rem;
  }
}

@media (max-width:360px) {
  .c1 {
    padding: 0px 1.6rem;
  }
}

<div>
  <section
    class="c0"
    id="#"
  >
    <div
      class="c1"
    >
      <p>
        Olá, seja bem-vindo ao meu portfólio.
      </p>
      <h1
        class="c2 c3"
      >
        Tício de Souza e Silva
        .
      </h1>
      <h2
        class="c2 c3"
      >
        Desenvolvedor WEB Júnior
        .
      </h2>
      <h2
        class="c2 c4"
      >
        Javascript | NodeJS | React
      </h2>
    </div>
  </section>
</div>
`;
